pkg/daemon/daemon.go

```diff
@@ -439,7 +441,19 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 	// When using systemd configuration we write the file
 	if vars.UsingSystemdMode {
 		log.Log.V(0).Info("nodeStateSyncHandler(): writing systemd config file to host")
-		systemdConfModified, err := systemd.WriteConfFile(dn.desiredNodeState)
+		// get node object
+		node := &corev1.Node{}
+		err := dn.client.Get(context.TODO(), client.ObjectKey{Name: vars.NodeName}, node)
+		if err != nil {
+			log.Log.Error(err, "nodeStateSyncHandler(): failed to get node object")
+			return err
+		}
+		netPoolConfig, _, err := utils.FindNodePoolConfig(context.Background(), node, dn.client)
+		if err != nil {
```

Comment:
I don't think this is the right way to handle it.

we can talk about it in the community meeting but in general my idea will be:

add a new section in the sriovNetworkNodeState.Spec.Config.Rdma Mode
add a new section in the sriovNetworkNodeState.Status.Config.Rdma Mode

so the configure daemon continues to only want to NodeState only

also in case of a change in the nodePool or selecting a different node pool for this node we will need to wait for a policy change so the reconcile will act on the nodePool change

WDYT?

Reply:
sounds good to me! I'll update my PR